app/services/openai-service.js

```diff
@@ -26,6 +27,40 @@ const generateEmbedding = async (chunk) => {
   return embedding[0]
 }
 
+/**
+ * Generates a short summary for a given text
+ * @param {string} text - The text to summarize
+ * @param {number} summaryTokenLimit - The token limit for the summary
+ * @returns {string} - The generated summary
+ */
+const generateShortSummary = async (text, summaryTokenLimit = 60) => {
+  const model = new ChatOpenAI({
+    azureOpenAIApiInstanceName: config.azureOpenAI.openAiInstanceName,
+    azureOpenAIApiKey: config.azureOpenAI.openAiKey,
+    azureOpenAIApiDeploymentName: 'gpt-35-turbo-16k',
+    azureOpenAIApiVersion: '2024-02-01',
+    onFailedAttempt
+  })
+
+  const messages = [
+    ['user', `
+      Generate a short summary of the following text:
+      ${text}
+    `]
+  ]
+
+  try {
+    const response = await model.generate(messages)
+
+    // extract the summary from the response
+    return response.generations.flat()[0].text.replace(/\n/g, ' ').trim()
```

Comment:
Would be nice if we can unit test this with an OpenAI mocked response

Reply:
I have created an elaborate try catch involving console.warn and .error. I have put in the tests that would catch the error and/or warn. The tests are passing. 
To take it further, how would you approach creating a mock test for the response?